libs/core/algorithms/tests/unit/algorithms/minmax_element.cpp

```diff
@@ -139,6 +140,7 @@ void minmax_element_test()
 {
     test_minmax_element<std::random_access_iterator_tag>();
     test_minmax_element<std::forward_iterator_tag>();
+    test_minmax_element_semantics<std::random_access_iterator_tag>();
```

Comment:
Don't you have to forward declare this function? Also, why not calling it for forward-iterators as well?